### PR TITLE
Misc makefile fixes

### PR DESCRIPTION
**Pull request by mchindri**

### CODE REVIEW

**Review by amiclaus — APPROVED** [review]
LGTM.
one question: were the makefile changes tested at least on one no-OS project (besides iio_demo)?

File: projects/iio_demo/src.mk [hunk 1]
```diff
@@ -27,12 +27,19 @@ SRC_DIRS += $(NO-OS)/network
 SRCS	 += $(NO-OS)/util/circular_buffer.c
 SRCS	 += $(PLATFORM_DRIVERS)/delay.c
 SRCS	 += $(PLATFORM_DRIVERS)/timer.c
+INCS	 += $(INCLUDE)/delay.h			\
```
Comment (buha) [inline projects/iio_demo/src.mk:30]: why were these files missing ? was this a hidden problem by the fact that the source files were not linked but all copied ?
if this is the case, then other projects may hide such problems too

**Review by adrimbarean — APPROVED** [review]
Looks good to me.

**Comment by buha** [comment]
Also, I think we need an easy way to generate a standalone project (with copied files). I think the current way would be 
`make LINK_SRCS=n` which is not easy to remember for users.

A rule like this one could make things easier:
`make standalone`

**Comment by mchindri** [comment]
> Also, I think we need an easy way to generate a standalone project (with copied files). I think the current way would be
> `make LINK_SRCS=n` which is not easy to remember for users.
> 
> A rule like this one could make things easier:
> `make standalone`

I'am working on that, but I have some issues with it. That is how I found the problem in iio_demo.
I will come with a PR just for that.

File: projects/iio_demo/src.mk (hunk 1 above)
Comment (mchindri) [reply projects/iio_demo/src.mk:30]: It was working  because coping src isn't well implemented. I will solve that in a pr for make standalone.

**Comment by mchindri** [comment]
> LGTM.
> one question: were the makefile changes tested at least on one no-OS project (besides iio_demo)?

yes

**Review by dbogdan — APPROVED** [review]

**Comment by dbogdan** [comment]
If another PR will implement "make standalone", we should be good to go with this one.

**Pull request merged**